lib/compress/zstd_compress.c

```diff
@@ -2566,6 +2567,26 @@ size_t ZSTD_getSequences(ZSTD_CCtx* zc, ZSTD_Sequence* outSeqs,
 
     ZSTD_compress2(zc, dst, dstCapacity, src, srcSize);
     ZSTD_customFree(dst, ZSTD_defaultCMem);
+
+    if (format == ZSTD_sf_noBlockDelimiters) {
+        /* Remove all block delimiters and append them to the next sequence's literals
+         * and do not emit last literals at all
```

Comment:
> and do not emit last literals at all

Well, even in `noBlockDelimiters` mode, there is still the issue of representing the last literals of the last block.

Reply:
In `noBlockDelimiters` mode we intentionally don't represent the last literals of the last block to be consistent, and then on the sequence compression side, I can still derive the last literals of the last block even without it, just based on how much of the source buffer we've consumed, right?

Reply:
Sure, that can work,
though it's very easy to miss for an API user, hence it should be clearly documented that there may be a few literals left in the buffer after the last sequence, that it's expected, and must be looked for.

To be fair, if it was me, I would only support `ZSTD_sf_explicitBlockDelimiters` for the extraction API,
as it's the most complete image of what's produced (`ZSTD_sf_noBlockDelimiters` throw information away, but doesn't bring any value).

If you worry about an ability to make a round-trip test with an ingestion API in "automated block delimiter mode", I would say this is a completely different topic. Such automated mode could still accept explicit block delimiter sequences, but would simply interpret them differently, merging them with following sequence.

Reply:
I agree that the last literals can be easily missed, so I'll make sure to clearly document this. I chose to not represent the last block last literals in `noBlockDelimiters` mode since I would like the sequences in our `ZSTD_Sequence` array to be as generic as possible, avoiding any "dummy" sequences - which is what the last literals would be if we were to represent them.

I still do kind of like the idea of allowing the specification of block delimiters or not for the extraction API - it enforces a stronger sense of symmetry between `getSequences()` and `compressSequences()`. I do agree that the `ZSTD_sf_noBlockDelimiters` only throws information away, but it seems like a nice feature to have if the user wanted to be generate sequences with both modes and compare them, in order to better visualize/understand the different expectations that `compressSequences()` might have depending on the mode.

As for testing, I will include a roundtrip unit test in my upcoming compress sequence API pull request, but I think the "real" test will come in a follow-up PR where I add a dedicated OSS Fuzzer that will generate arbitrary sequences on a literal buffer (not using the `ZSTD_getSequences()` API), and try to compress and decompress those arbitrary sequences.